src/Qir/Controller/test-qir-controller.ps1

```diff
@@ -19,23 +20,38 @@ Foreach-Object {
     # Get the paths to the output and error files to pass to the QIR controller.
     $outputFile = (Join-Path $testArtifactsFolder ($_.BaseName + ".out"))
     $errorFile = (Join-Path $testArtifactsFolder ($_.BaseName + ".err"))
-    dotnet run --project $controllerProject -- --input $_.FullName --output $outputFile --error $errorFile
+    dotnet run --project $controllerProject -- --input $_.FullName --output $outputFile --error $errorFile --includeDirectory "placeholder for now" --libraryDirectory "placeholder for now"
 
     # Compare the expected content of the output and error files vs the actual content.
     $expectedOutputFile = (Join-Path $testCasesFolder ($_.BaseName + ".out"))
-    $expectedOutput = Get-Content -Path $expectedOutputFile -Raw
-    $actualOutput = Get-Content -Path $outputFile -Raw
-    if (-not ($expectedOutput -ceq $actualOutput)) {
-        Write-Host "##vso[task.logissue type=error;]Failed QIR Controller test case: $($_.BaseName)"
-        Write-Host "##[info]Expected output:"
-        Write-Host $expectedOutput
-        Write-Host "##[info]Actual output:"
-        Write-Host $actualOutput
-        $script:all_ok = $False
-        break
+    $expectedErrorFile = (Join-Path $testCasesFolder ($_.BaseName + ".err"))
+
+    if ((Test-Path $expectedOutputFile)) {
+        $expectedOutput = Get-Content -Path $expectedOutputFile -Raw
+        $actualOutput = Get-Content -Path $outputFile -Raw
+        if (-not ($expectedOutput -ceq $actualOutput)) {
+            Write-Host "##vso[task.logissue type=error;]Failed QIR Controller test case: $($_.BaseName)"
+            Write-Host "##[info]Expected output:"
+            Write-Host $expectedOutput
+            Write-Host "##[info]Actual output:"
+            Write-Host $actualOutput
+            $script:all_ok = $False
+        }
+        elseif ((Test-Path $errorFile)) {
```

Comment:
Just noticed this. The existence of an output files does not mean an error did not occur. A simulation can fail at any point but it could still output something before that, which is important to debug where it failed.

We could have two independent `if` statements (one for the error file and one for the output file) and the logic inside should only check for the contents of that particular file.

Reply:
Got it, I think I misunderstood something in the spec.